### ISSUE TITLE
End coordinate for short sequences

### ISSUE THREAD

**Issue by muffato** [tolkit/fasta_windows#9]
Hi Max,

Follow-up of #8. I've found the same issue again, but this time in sequences that are shorter than the window size.
I believe this is caused by the fact the first window is always set to end at the position `window_size`, cf https://github.com/tolkit/fasta_windows/blob/master/src/fw.rs#L75-L96 
The fix you put in place only kicks in later (line 122).

Cheers,
Matthieu

**Comment by Euphrasiologist** [comment]
Hi Matthieu, 

Thanks for raising this - I will get to this and push it onto conda again. Let's see what I can do.

Thanks!
M

**Comment by muffato** [comment]
Hi Max. Thank you very much ! I've tested it and can confirm this is fixed.

However, compared to a previous run, the values for the di-, tri-, and tetra-nucleotide Shannon diversity have changed. Is that expected ?

**Comment by Euphrasiologist** [comment]
Was a bit panicked, then I remember I changed the log function to a base of 2 (from natural log).

https://github.com/tolkit/fasta_windows/blob/ba8f622570f0476343e413ed624d99b344bce265/src/kmeru8.rs#L111

This will account for the differences. I changed this because other folk using shannon entropy on kmers also use base 2. I'm still a little unsure the implementation of Shannon Entropy is correct, so that may be changed in the future.
